File: Ejercicio/1.py
class Videojuego:
    def __init__(self, titulo, precio, lanzamiento, plataforma=None, peso=None):
        self.titulo = titulo
        self.precio = precio
        self.lanzamiento = lanzamiento
        self.plataforma = plataforma
        self.peso = peso

    def mostrarInfo(self):
        print("Titulo:", self.titulo)
        print("Precio: S/. ", self.precio)
        print("Lanzamiento:", self.lanzamiento)

    def mostrarInfoDigital(self):
        print("Plataforma:", self.plataforma)

    def mostrarInfoFisico(self):
        print("Peso:", self.peso, "kg")


def mostrarMenu():
    print("Seleccione una opcion: ")
    print("1) Mostrar info del videojuego digital")
    print("2) Mostrar info del videojuego fisico")
    print("3) Salir")


juegoDigital01 = Videojuego("Half-Life 2", 22.00, 2004, "Steam")
juegoFisico01 = Videojuego("Outlast 2", 50.00, 2017, peso=0.2)

opcion = 0
while opcion != 3:
    mostrarMenu()
    opcion = int(input())
    if opcion == 1:
        print("Informacion del videojuego digital: ")
        juegoDigital01.mostrarInfo()
        juegoDigital01.mostrarInfoDigital()
    elif opcion == 2:
        print("Informacion del videojuego fisico: ")
        juegoFisico01.mostrarInfo()
        juegoFisico01.mostrarInfoFisico()
    elif opcion == 3:
        print("Saliendo...")
    else:
        print("Opcion invalida, intentelo de nuevo")

File: Ejercicio/1.cpp
#include <iostream>
#include <string>
#include <map> // Necesario para usar mapas en C++
using namespace std;

class Videojuego 
{
    private:
        string titulo;
        float precio;
        int lanzamiento;
    public:
        // Constructor para juego digital
        Videojuego(string _titulo, float _precio, int _lanzamiento, string _plataforma) 
        {
            titulo = _titulo;
            precio = _precio;
            lanzamiento = _lanzamiento;
            plataforma = _plataforma;
        }

        // Constructor para juego fisico
        Videojuego(string _titulo, float _precio, int _lanzamiento, float _peso) 
        {
            titulo = _titulo;
            precio = _precio;
            lanzamiento = _lanzamiento;
            peso = _peso;
        }


        void mostrarInfo() 
        {
            cout << "Titulo: " << titulo << endl;
            cout << "Precio: S/. " << precio << endl;
            cout << "Lanzamiento: " << lanzamiento << endl;
        }

        // Metodos adicionales para atributos especiales
        void mostrarInfoDigital() 
        {
            cout << "Plataforma: " << plataforma << endl;
        }

        void mostrarInfoFisico() 
        {
            cout << "Peso: " << peso << " kg" << endl;
        }

    private:
        string plataforma; // Solo para juegos digitales
        float peso; // Solo para juegos fisicos
};

void mostrarMenu() 
{
    cout << "Seleccione una opcion: " << endl;
    cout << "1) Mostrar info del videojuego digital" << endl;
    cout << "2) Mostrar info del videojuego fisico" << endl;
    cout << "3) Salir." << endl;
}

int main() 
{
    Videojuego juegoDigital01("Half-Life 2", 22.00, 2004, "Steam");
    Videojuego juegoFisico01("Outlast 2", 50.00, 2017, 0.2);

    int opcion;
   do {
        mostrarMenu();
        cin >> opcion;
        switch (opcion) 
        {
            case 1:
                cout << "Informacion del videojuego digital:" << endl;
                juegoDigital01.mostrarInfo();
                juegoDigital01.mostrarInfoDigital();
                break;
            case 2:
                cout << "Informacion del videojuego fisico:" << endl;
                juegoFisico01.mostrarInfo();
                juegoFisico01.mostrarInfoFisico();
                break;
            case 3:
                cout << "Saliendo..." << endl;
                break;
            default:
                cout << "Opcion invalida. Intentelo de nuevo." << endl;
                break;
        }
    } while (opcion != 3);


    return 0;
}
